ticketbuyer/buyer.go

```diff
@@ -104,6 +101,198 @@ type TicketPurchaser struct {
+		TicketAddress:             t.cfg.TicketAddress,
+		TxFee:                     t.cfg.TxFee,
+	}
+	t.purchaserMtx.Unlock()
+	return config
+}
+
+// AccountName returns the account name to use for purchasing tickets.
+func (t *TicketPurchaser) AccountName() string {
+	t.purchaserMtx.Lock()
+	accountName := t.accountName
+	t.purchaserMtx.Unlock()
+	return accountName
+}
+
+// SetAccountName sets the account name to use for purchasing tickets.
+func (t *TicketPurchaser) SetAccountName(accountName string) {
+	t.purchaserMtx.Lock()
+	t.accountName = accountName
+	t.purchaserMtx.Unlock()
+}
+
+// BalanceToMaintain returns the balance to be maintained in the wallet.
+func (t *TicketPurchaser) BalanceToMaintain() dcrutil.Amount {
+	t.purchaserMtx.Lock()
+	balanceToMaintain := t.balanceToMaintain
+	t.purchaserMtx.Unlock()
+	return balanceToMaintain
+}
+
+// SetBalanceToMaintain sets the balance to be maintained in the wallet.
+func (t *TicketPurchaser) SetBalanceToMaintain(balanceToMaintain dcrutil.Amount) {
+	t.purchaserMtx.Lock()
+	t.balanceToMaintain = balanceToMaintain
+	t.purchaserMtx.Unlock()
+}
+
+// MaxPriceAbsolute returns the max absolute price to purchase a ticket.
+func (t *TicketPurchaser) MaxPriceAbsolute() dcrutil.Amount {
+	t.purchaserMtx.Lock()
+	maxPriceAbsolute := t.maxPriceAbsolute
+	t.purchaserMtx.Unlock()
+	return maxPriceAbsolute
+}
+
+// SetMaxPriceAbsolute sets the max absolute price to purchase a ticket.
+func (t *TicketPurchaser) SetMaxPriceAbsolute(maxPriceAbsolute dcrutil.Amount) {
+	t.purchaserMtx.Lock()
+	t.maxPriceAbsolute = maxPriceAbsolute
+	t.purchaserMtx.Unlock()
+}
+
+// MaxPriceRelative returns the scaling factor which is multipled by average
+// price to calculate a relative max for the ticket price.
+func (t *TicketPurchaser) MaxPriceRelative() float64 {
+	t.purchaserMtx.Lock()
+	maxPriceRelative := t.maxPriceRelative
+	t.purchaserMtx.Unlock()
+	return maxPriceRelative
+}
+
+// SetMaxPriceRelative sets scaling factor.
+func (t *TicketPurchaser) SetMaxPriceRelative(maxPriceRelative float64) {
+	t.purchaserMtx.Lock()
+	t.maxPriceRelative = maxPriceRelative
+	t.purchaserMtx.Unlock()
+}
+
+// MaxFee returns the max ticket fee per KB to use when purchasing tickets.
+func (t *TicketPurchaser) MaxFee() dcrutil.Amount {
+	t.purchaserMtx.Lock()
+	maxFee := t.maxFee
+	t.purchaserMtx.Unlock()
+	return maxFee
+}
+
+// SetMaxFee sets the max ticket fee per KB to use when purchasing tickets.
+func (t *TicketPurchaser) SetMaxFee(maxFee dcrutil.Amount) {
+	t.purchaserMtx.Lock()
+	t.maxFee = maxFee
+	t.purchaserMtx.Unlock()
+}
+
+// TicketAddress returns the address to send ticket outputs to.
+func (t *TicketPurchaser) TicketAddress() dcrutil.Address {
+	t.purchaserMtx.Lock()
+	ticketAddress := t.ticketAddress
+	t.purchaserMtx.Unlock()
+	return ticketAddress
+}
+
+// SetTicketAddress sets the address to send ticket outputs to.
+func (t *TicketPurchaser) SetTicketAddress(ticketAddress string) error {
```

Comment:
take the decoded address not a string.  the rpc server should do all conversions and return the correct errors for invalid parameters.